tests/L0/run_optimizers/test_fused_optimizer.py

```diff
@@ -50,41 +51,54 @@ def get_max_diff(self, ref_param, tst_param):
 
         return max_abs_diff, max_rel_diff
 
-    def gen_single_type_test(self, param_type=torch.float):
+    def gen_single_type_test(self, param_type=torch.float, device='cuda'):
         nelem = 278011
-        adam_option = {'lr':5e-4, 'betas':(0.9, 0.999), 'eps':1e-08,
-            'weight_decay':0, 'amsgrad':False}
 
-        tensor = torch.rand(nelem, dtype=param_type, device='cuda')
+        tensor = torch.rand(nelem, dtype=param_type, device=device)
         ref_param, tst_param, ref_optim, tst_optim = \
-            self.gen_param_optim([tensor], adam_option)
+            self.gen_param_optim([tensor], self.options)
 
         for i in range(self.iters):
             self.gen_grad(ref_param, tst_param)
             ref_optim.step()
             tst_optim.step()
             max_abs_diff, max_rel_diff = self.get_max_diff(ref_param, tst_param)
-
             self.assertLessEqual(max_abs_diff, self.max_abs_diff)
             self.assertLessEqual(max_rel_diff, self.max_rel_diff)
 
+
+class TestFusedAdam(TestFusedOptimizer):
+
+    def __init__(self, *args, **kwargs):
+        super(TestFusedAdam, self).__init__(*args, **kwargs)
+        self.options = {'lr':5e-4, 'betas':(0.9, 0.999), 'eps':1e-08,
+            'weight_decay': 0, 'amsgrad': False}
+        self.ref_optim = torch.optim.Adam
+        self.fused_optim = apex.optimizers.FusedAdam
+
     def test_float(self):
         self.gen_single_type_test(param_type=torch.float)
 
     def test_half(self):
         self.gen_single_type_test(param_type=torch.float16)
 
+    @unittest.skipIf(torch.cuda.device_count()<2, "more than 1 GPU required")
+    def test_multi_device(self):
+        devices = ("cuda:0", "cuda:1")
+        for current_dev, tensor_dev in product(devices, devices):
+            with torch.cuda.device(current_dev):
+                self.gen_single_type_test(param_type=torch.float, device=tensor_dev)
```

Comment:
I like this test.  Can you add it at the end of [test_lamb.py](https://github.com/NVIDIA/apex/blob/5b53121a2124d70dad0fd4f462d1392d06573ca9/tests/L0/run_optimizers/test_lamb.py#L255) as well?  Don't bother merging test_lamb.py into this file, that would take more effort than it's worth.